react/OrderItems.tsx

```diff
@@ -254,11 +249,12 @@ const useAddItemsTask = (
         return updatedOrderForm
       },
       rollback: () => {
-        setOrderForm(prevOrderForm => {
+        setOrderForm((prevOrderForm) => {
           const itemIds = mutationInputItems.map(({ id }) => id!.toString())
+
           return {
             ...prevOrderForm,
-            items: prevOrderForm.items.filter(orderFormItem => {
+            items: prevOrderForm.items.filter((orderFormItem) => {
               return !itemIds.includes(orderFormItem.id)
             }),
```

Comment:
since this task now consist of both new items (i.e. item that didn't exist in the cart before this operation) and updated items (items that had it's quantity changed), you'll need to add the logic here to rollback to the previous quantity of the item before this mutation happened. you can get some inspiration in the `rollback` function of the `updateItems` task

Reply:
I think that you could split the `addItems` call in two function calls, to avoid duplicating the rollback logic of `updateItemsTask` here, one call to actually add the new items in the cart (which could proceed normally in the addItems function) and another call to the `updateQuantity` that would increase the quantity of the items that already exists in the cart. What do you think?